odilo.py

```diff
@@ -0,0 +1,598 @@
+        ),
+        "JPG": (
+            Representation.JPEG_MEDIA_TYPE, DeliveryMechanism.NO_DRM
+        ),
+        "SCORM": (
+            Representation.SCORM_MEDIA_TYPE, DeliveryMechanism.NO_DRM
+        )
+    }
+
+    odilo_medium_to_simplified_medium = {
+        "ACSM": Edition.BOOK_MEDIUM,
+        "EBOOK_STREAMING": Edition.BOOK_MEDIUM,
+        "MP3": Edition.AUDIO_MEDIUM,
+        "MP4": Edition.VIDEO_MEDIUM,
+        "WMV": Edition.VIDEO_MEDIUM,
+        "JPG": Edition.IMAGE_MEDIUM,
+        "SCORM": Edition.COURSEWARE_MEDIUM
+    }
+
+    @classmethod
+    def record_info_to_circulation(cls, availability):
+        """ Note:  The json data passed into this method is from a different file/stream
+        from the json data that goes into the record_info_to_metadata() method.
+        """
+
+        if 'recordId' not in availability:
+            return None
+
+        record_id = availability['recordId']
+        primary_identifier = IdentifierData(Identifier.ODILO_ID, record_id)  # We own this availability.
+
+        licenses_owned = int(availability['totalCopies'])
+        licenses_available = int(availability['availableCopies'])
+
+        # 'licenses_reserved' is the number of patrons who put the book on hold earlier,
+        #  but who are now at the front of the queue and who could get the book right now if they wanted to.
+        if 'notifiedHolds' in availability:
+            licenses_reserved = int(availability['notifiedHolds'])
+        else:
+            licenses_reserved = 0
+
+        # 'patrons_in_hold_queue' contains the number of patrons who are currently waiting for a copy of the book.
+        if 'holdsQueueSize' in availability:
+            patrons_in_hold_queue = int(availability['holdsQueueSize'])
+        else:
+            patrons_in_hold_queue = 0
+
+        return CirculationData(
+            data_source=DataSource.ODILO,
+            primary_identifier=primary_identifier,
+            licenses_owned=licenses_owned,
+            licenses_available=licenses_available,
+            licenses_reserved=licenses_reserved,
+            patrons_in_hold_queue=patrons_in_hold_queue,
+        )
+
+    @classmethod
+    def image_link_to_linkdata(cls, link, rel):
+        if not link:
+            return None
+
+        return LinkData(rel=rel, href=link, media_type=Representation.JPEG_MEDIA_TYPE)
+
+    @classmethod
+    def record_info_to_metadata(cls, book, availability):
+        """Turn Odilo's JSON representation of a book into a Metadata
+        object.
+
+        Note:  The json data passed into this method is from a different file/stream
+        from the json data that goes into the book_info_to_circulation() method.
+        """
+        if 'id' not in book:
+            return None
+
+        odilo_id = book['id']
+        primary_identifier = IdentifierData(Identifier.ODILO_ID, odilo_id)
+        active = book.get('active')
```

Comment:
What does 'active' mean in this context? I noticed that you won't treat a metadata import as successful unless `active` is true. If a book is not 'active' because its metadata is incomplete then that's appropriate. If a book is not 'active' because it's no longer in the collection, then it would be better to allow the import to proceed but make sure the `CirculationData` makes it clear that the work is not available.

Reply:
"active" means that the book it's no longer in the collection, so, I have corrected that as you advised me.